### PR TITLE
Improvements related to liveness napp

### PR DESCRIPTION
**Pull request by diraol**
Here we have some improvements on `Kyco` related to the `liveness` core napp.

Some changes were made to allow the development of the napp, and others are related to some fixes of bugs found after the implementation of the napp.

Some of the fixes are also related to the issue #58 also.

CHANGELOG:
- Improvements on client connection errors and exceptions handling
- Compatibilization with the liveness core napp

### CODE REVIEW

File: kyco/core/tcp_server.py [hunk 1]
```diff
@@ -107,9 +109,18 @@ def handle(self):
...
                                                connection_id=connection_id)
                 self.server.controller.buffers.raw.put(event)
-        except Exception:
+        except (SocketError, OSError) as exception:
+            # TODO: Client disconnected is the only possible reason?
             log.info("Client %s:%s disconnected", self.ip, self.port)
-            raise Exception
+            error_content = {'destination': (self.ip, self.port),
+                             'exception': exception, 'event': None}
+            event = KycoMessageOutError(content=error_content)
```
Comment (beraldoleal) [inline kyco/core/tcp_server.py]: Rename KycoMessageOutError to KycoError or something like that

File: kyco/core/exceptions.py [hunk 2]
```diff
@@ -3,6 +3,21 @@
...
+    def __str__(self):
+        return 'KycoCore exception: ' + super().__str__()
+
+
+class KycoSwitchOfflineException(Exception):
+    def __init__(self, dpid):
+        self.dpid = dpid
+
+    def __str__(self):
+        msg = 'The switch {} is offline, you can not send a message to it'
```
Comment (beraldoleal) [inline kyco/core/exceptions.py]: Please, change this message to "The switch {} is not reachable. Please check the connection between the switch and the controller."

**Review by beraldoleal — CHANGES_REQUESTED** [review]

File: kyco/core/tcp_server.py (hunk 1 above)
Comment (diraol) [reply kyco/core/tcp_server.py]: Issue addressed on a191452

File: kyco/core/exceptions.py (hunk 2 above)
Comment (diraol) [reply kyco/core/exceptions.py]: Issue addressed on 6dc2f7e

**Pull request closed without merging**